src/cli/serve/deprecated_config.js

```diff
@@ -0,0 +1,16 @@
+import { forOwn, has, noop } from 'lodash';
+
+// deprecated settings are still allowed, but will be removed at a later time. They
+// are checked for after the config object is prepared and known, so legacySettings
+// will have already been transformed.
+export const deprecatedSettings = new Map([
+  [['server', 'xsrf', 'token'], 'server.xsrf.token is deprecated. It is no longer used when providing xsrf protection.']
```

Comment:
Ah, I didn't see that test.  Can you explain what "ignoring" does in this context?  Maybe my brain is just fried this late in the day, but I'm struggling to make sense of it.  Why would the "compoundedness" of the key determine whether a configuration should be considered deprecated?